ESP32-BonadroneFC/ESP32-BonadroneFC.ino

```diff
@@ -0,0 +1,158 @@
+#include <WiFi.h>
+
+// The ESP32 default IP address is 192.168.4.1
+// To set a different static IP see:
+// https://github.com/espressif/arduino-esp32/blob/master/libraries/WiFi/examples/WiFiClientStaticIP/WiFiClientStaticIP.ino
+
+// password should be equal or longer than 8 chars
+// for enabling this configuration
+const char *ssid = "Bonadrone";
+const char *password = "12345678";
+
+// Set web server port number to 80
+WiFiServer server(80);
+
+// Battery
+typedef union
+{
+  float number;
+  uint8_t bytes[4];
+} floatUnion_t;
+const int BATTERY_PIN = 33; // specific to BonaDrone FC board
+floatUnion_t batteryVoltage;
+
+
+
+void measureBatteryVoltage()
+{
+  const float voltageDivider = 1.15;
+  
+  static float batLast = 0.0;
+  static float batTrend = 0.0;
+  static int ii = 0;
+  
+  float sFact_1 = 0.2;
+  float sFact_2 = 0.8;
+  
+  int meas = analogRead(33);
+  float batMeas = (float)meas/4096.0*12.60;
```

Comment:
It might be better to define the values of `4096.0` and `12.60` as constants instead of having them as hard-coded values.